tasks/GroundFollower.cpp

```diff
@@ -0,0 +1,148 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "GroundFollower.hpp"
+#include <base-logging/Logging.hpp>
+
+using namespace auv_control;
+
+GroundFollower::GroundFollower(std::string const& name, TaskCore::TaskState initial_state)
+    : GroundFollowerBase(name, initial_state)
+{
+}
+
+GroundFollower::GroundFollower(std::string const& name, RTT::ExecutionEngine* engine, TaskCore::TaskState initial_state)
+    : GroundFollowerBase(name, engine, initial_state)
+{
+}
+
+GroundFollower::~GroundFollower()
+{
+}
+
+
+
+/// The following lines are template definitions for the various state machine
+// hooks defined by Orocos::RTT. See GroundFollower.hpp for more detailed
+// documentation about them.
+
+bool GroundFollower::configureHook()
+{
+    if (! GroundFollowerBase::configureHook())
+        return false;
+
+    new_altimeter_timeout = base::Timeout(_altimeter_timeout.get());
+    new_depth_timeout = base::Timeout(_depth_timeout.get());
+    altimeter_dropout_timeout = base::Timeout(_altimeter_dropout_timeout.get());
+
+    last_valid_ground_position = NAN;
```

Comment:
I would move the invalidation to the start hook and use `base::unknown<double>()` which uses `std::numeric_limits`.